### PR TITLE
Added ability to specify value on list and group-list selections

### PR DESCRIPTION
**Pull request by OmenApps**
- Updated Select2ListView, Select2GroupListView, & Select2ListChoiceField to account for the extra functionality
- Added tests
- Updated docs to reflect

Have successfully run the tests, and implemented multiple versions of the resulting view & form variations in my own project.

### CODE REVIEW

**Comment by OmenApps** [comment]
Also, just noticed this should close out #985

File: docs/tutorial.rst [hunk 1]
```diff
@@ -737,9 +737,42 @@ make it easier to avoid problems when using Select2ListView. For example:
...
+
+
+To define your own values for each selection, provide a list-of-lists or
+list-of-tuples for the Select2ListView choice_list, and use Select2ListView
+instead of Select2ListView. For example:
+
+.. code-block:: python
+
+    class CountryAutocompleteFromList(autocomplete.Select2ListView):
+    def get_list(self):
```
Comment (jpic) [inline docs/tutorial.rst:751]: This might be missing an identation level

Comment (OmenApps) [reply docs/tutorial.rst:751]: Will fix this and the ci identified issue this evening.

Comment (jpic) [reply docs/tutorial.rst:751]: Nice, not much to say here, it's really a quality contribution you got right there, with tests and documentation :tophat:

**Comment by jpic** [comment]
This passes all tests locally, the GFK test requires master version of django-querysetsequence but that is unrelated.

Merging, thank you !!!

**Pull request merged**